joint_limits_enforcement_plugins/test/test_ruckig_joint_limiter.cpp

```diff
@@ -12,33 +12,276 @@
+  if (joint_limiter_)
+  {
+    Init();
+    Configure();
 
-  std::unique_ptr<JointLimiter> joint_limiter;
-  std::string joint_limiter_type = "ruckig_joint_limiter/RuckigJointLimiter";
+    rclcpp::Duration period(0, 5000000);  // 0.005 second AS HARDCODED in the lib
+    // pos, vel, acc, dec = 1.0, 2.0, 5.0, 7.5
 
-  joint_limiter =
-    std::unique_ptr<JointLimiter>(joint_limiter_loader.createUnmanagedInstance(joint_limiter_type));
-  ASSERT_NO_THROW(
-    joint_limiter = std::unique_ptr<JointLimiter>(
-      joint_limiter_loader.createUnmanagedInstance(joint_limiter_type)));
-  ASSERT_NE(joint_limiter, nullptr);
+    // within limits
+    desired_joint_states_.positions[0] = 1.0;
+    desired_joint_states_.velocities[0] = 0.01;
+    desired_joint_states_.accelerations[0] = 2.0;
+    ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+    // check if max jerk applied
+    CHECK_STATE_SINGLE_JOINT(
+      desired_joint_states_, 0,
+      desired_joint_states_.positions[0],     // pos unchanged
+      desired_joint_states_.velocities[0],    // vel unchanged
+      desired_joint_states_.accelerations[0]  // acc unchanged
+    );
+  }
+}
+
+TEST_F(RuckigJointLimiterTest, when_velocity_exceeded_expect_vel_and_acc_enforced)
+{
+  SetupNode("ruckig_joint_limiter");
+  Load();
+
+  if (joint_limiter_)
+  {
+    Init();
+    Configure();
+
+    rclcpp::Duration period(0, 5000000);  // 0.005 second AS HARDCODED in the lib
+
+    // desired velocity exceeds
+    desired_joint_states_.positions[0] = 1.0;
+    desired_joint_states_.velocities[0] = 1.0;
+    ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+    // check if vel and acc limits applied
+    CHECK_STATE_SINGLE_JOINT(
+      desired_joint_states_, 0,
+      desired_joint_states_.positions[0],  // pos unchanged
+      // TODO(gwalck) why is the result divided by 2.0 ?
+      desired_joint_states_.accelerations[0] * period.seconds() /
+        2.0,                    // vel limited to max_acc * dt / 2.0
+      100.0 * period.seconds()  // acc set to max_jerk * dt
+    );
+
+    // check opposite velocity direction (sign copy)
+    desired_joint_states_.positions[0] = -1.0;
+    desired_joint_states_.velocities[0] = -1.0;
+    desired_joint_states_.accelerations[0] = -1.0;
+    ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+    // check if vel and acc limits applied
+    CHECK_STATE_SINGLE_JOINT(
+      desired_joint_states_, 0,
+      desired_joint_states_.positions[0],  // pos unchanged
+      // TODO(gwalck) why is the result divided by 2.0 ?
+      desired_joint_states_.accelerations[0] * period.seconds() /
+        2.0,                     // vel limited to max_acc * dt / 2.0
+      -100.0 * period.seconds()  // acc set to max_jerk * dt
+    );
+  }
+}
+
+/* no pos enforcing available in ruckig
```

Comment:
we can leave here and expect ruckig to fail with those values. Then we know that we handle those situations properly.

Reply:
> we can leave here

I suppose you mean we can activate this test and expect the limiter to NOT limit.